server/src/main/java/org/elasticsearch/action/fieldcaps/RequestDispatcher.java

```diff
@@ -56,15 +59,15 @@ final class RequestDispatcher {
     private final OriginalIndices originalIndices;
     private final long nowInMillis;
 
-    private final boolean hasFilter;
+    private final boolean withIndexFilter;
```

Comment:
I think it'd be helpful to describe the execution strategy in the  javadoc for the `RequestDispatcher` class, because it would really helps others in understanding this logic. We could summarize how we deduplicate based on mapping hash, and how we need to disable the optimizations when there's an index filter.

Reply:
+1, I added a javadocs in https://github.com/elastic/elasticsearch/pull/84598/commits/886893e0654bf7f79376b04f8a4d70d6cd433afa